fe/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java

```diff
@@ -0,0 +1,229 @@
+        if (attachment instanceof BrokerPendingTaskAttachment) {
+            onPendingTaskFinished((BrokerPendingTaskAttachment) attachment);
+        } else if (attachment instanceof BrokerLoadingTaskAttachment) {
+            onLoadingTaskFinished((BrokerLoadingTaskAttachment) attachment);
+        }
+    }
+
+    @Override
+    public void onTaskFailed(String errMsg) {
+        updateState(JobState.CANCELLED, FailMsg.CancelType.LOAD_RUN_FAIL, errMsg);
+    }
+
+    /**
+     * step1: divide job into loading task
+     * step2: init the plan of task
+     * step3: submit tasks into loadingTaskExecutor
+     * @param attachment BrokerPendingTaskAttachment
+     */
+    public void onPendingTaskFinished(BrokerPendingTaskAttachment attachment) {
+        // TODO(ml): check if task has been cancelled
+        Database db = null;
+        try {
+            getDb();
+        } catch (MetaNotFoundException e) {
+            LOG.warn(new LogBuilder(LogKey.LOAD_JOB, id)
+                             .add("database_id", dbId)
+                             .add("error_msg", "Failed to divide job into loading task when db not found.")
+                             .build(), e);
+            updateState(JobState.CANCELLED, FailMsg.CancelType.ETL_RUN_FAIL, "db does not exist. id: " + dbId);
+            return;
+        }
+
+        // divide job into broker loading task by table
+        db.readLock();
+        try {
+            for (Map.Entry<Long, List<BrokerFileGroup>> entry :
+                    dataSourceInfo.getIdToFileGroups().entrySet()) {
+                long tableId = entry.getKey();
+                OlapTable table = (OlapTable) db.getTable(tableId);
+                if (table == null) {
+                    LOG.warn(new LogBuilder(LogKey.LOAD_JOB, id)
+                                     .add("database_id", dbId)
+                                     .add("table_id", tableId)
+                                     .add("error_msg", "Failed to divide job into loading task when table not found")
+                                     .build());
+                    updateState(JobState.CANCELLED, FailMsg.CancelType.ETL_RUN_FAIL,
+                                "Unknown table(" + tableId + ") in database(" + db.getFullName() + ")");
+                    return;
+                }
+
+                // Generate loading task and init the plan of task
+                LoadLoadingTask task = new LoadLoadingTask(db, table, brokerDesc,
+                                                           entry.getValue(), getDeadlineMs(), execMemLimit,
+                                                           transactionId, this);
+                task.init(attachment.getFileStatusByTable(tableId),
+                          attachment.getFileNumByTable(tableId));
+                // Add tasks into list and pool
+                tasks.add(task);
+                Catalog.getCurrentCatalog().getLoadManager().submitTask(task);
```

Comment:
Can you pass loadManager when this object is created?
we should avoid using getInstance() in our code